text/0061-defrag-memos.md

```diff
@@ -0,0 +1,68 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal adds one new encrypted memo type, the Defragmentation Memo. Defragmentation transactions will write this memo to the payload `TxOut`. Defragmentation transactions will also not produce a change `TxOut`, as there should never be change and it is not needed to construct RTH. This memo has type bytes `0x0002`.
+
+The Defragmentation Memo contains three pieces of data. Like RTH's Destination Memo, the fee and total outlay of the transaction are written to the memo. In addition to this, there is also a defragmentation ID number. If a highly fragmented account needs to send a large amount, it may require more than one transaction to sufficiently defragment. The defragmentation ID number can be used by clients to group defragmentation transactions from one session together. This might be desirable in order to show how much was paid in fees for defragmentation. If used, the defragmentation ID number should be selected randomly.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## 0x0002 Defragmentation Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 7      | Big-endian bytes of fee amount, an unsigned 56-bit number |
+| 7 - 15     | Big-endian bytes of the total outlay amount, an unsigned 64-bit integer |
+| 15 - 23    | Big-endian bytes of the defragmentation transaction ID, an unsigned 64-bit integer |
+| 23 - 64    | Unused bytes |
+
+The fee amount is stored as a u56 in the first 7 bytes. This is done for consistency with RTH Destination memos. The next 8 bytes contain the total outlay of the defragmentation transaction. Lastly, the defragmentation ID is stored in 8 bytes as a u64.
+
+This memo will be written to the payload `TxOut` of a defragmentation transaction. Defragmentation transactions should not produce a change `TxOut`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Accounts which have already performed defragmentation will either have no memos or sender/destination memos for their old defragmentation transactions. Though we encourage not using an account with more than one application, not all developers will implement protections against it nor will all users heed this advice. If new applications do not handle legacy defragmentation transactions, they may inadvertently display them as normal transactions.
```

Comment:
> Though we encourage not using an account with more than one application

This makes sense but where/when did we decide this, and what is the rationale.

Reply:
Well, if we just think about the difference between current desktop and mobile wallets, if you have MOB sent to your desktop wallet address, a Fog client won't be able to find the TxOut. So that's one incompatibility. But consider also that all clients (desktop or mobile) may want/need to store info on their own back-end to support additional payment app features. MobileCoin provides, for example, memo types for payment requests and intents. But no such thing as a payment intent or request exists on the blockchain. To make use of these features, client applications will need to use some third-party back-end with info explaining the details of those requests/intents. It wouldn't make sense that different clients from different developers would have knowledge of each other's payment requests/intents.

Reply:
The rationale seems reasonable but "Though we" in this context implies MobileCoin has taken a public position, yet in some cases we've recommended people load their Signal entropy into Moby. Its a nit-pick but I don't think we should refer to the authority of company position unless its been made public.

Reply:
Yeah, I agree. The word "we" should not have been used here and I will update this paragraph

Reply:
So, to expand on this, what we encourage is not using the same account across different Fog Servers, or the absence of one. This is to prevent the situation where a Fog user ends up missing txos. A Desktop Wallet account _can_ be made to support Fog credentials (supported through Full Service), which would make it possible to have the account loaded and useable with no lost information from both instances.